### PR TITLE
Put password character limit to 128

### PR DESCRIPTION
**Pull request by k9845**

### CODE REVIEW

**Review by annshress — APPROVED** [review]

**Comment by thenav56** [comment]
@k9845 Please go through comments from the old PR
https://github.com/the-deep/server/pull/613

File: apps/user/tests/test_password.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class PasswordCheckerTest(TestCase):
+
+    def test_password_greater_than_128_characters(self):
+        expected_error = "This password has exceed the limit of %d characters"
+        self.assertIsNone(CustomMaximumLengthValidator().validate('12345678'))
+        self.assertIsNone(CustomMaximumLengthValidator(max_length=20).validate('123'))
+
+        with self.assertRaises(ValidationError) as vd:
+            CustomMaximumLengthValidator(max_length=128).validate('12'*129)
+        self.assertEqual(vd.exception.messages, [expected_error % 128])
```
Comment (annshress) [inline apps/user/tests/test_password.py]: Message can change with time. Lets check against `exception.code` instead.

**Comment by annshress** [comment]
@k9845 Password limit for login is WIP? Or is it going to have a separate PR?

File: apps/jwt_auth/serializers.py [hunk 2]
```diff
@@ -24,6 +24,11 @@ def validate_recaptcha(self, recaptcha_response):
         if not validate_recaptcha(recaptcha_response):
             raise InvalidCaptchaError
 
+    def validate_password(self, value):
+        password_validation.validate_password(value, self.instance)
+        print(value)
```
Comment (annshress) [inline apps/jwt_auth/serializers.py]:
```suggestion
```

File: apps/jwt_auth/tests/test_apis.py [hunk 3]
```diff
@@ -37,3 +38,15 @@ def test_refresh_token(self):
...
+        data = {
+            'username': "Hari@gmail.com",
+            "password": 'a'*130
+        }
+        url = '/api/v1/token/'
+
+        response = self.client.post(url, data)
+        self.assert_400(response)
+        assert 'password' in response.data['errors']
+        self.assertEqual(response.data['errors']['password'][0], 'This password has exceed the limit of 128 characters')
```
Comment (annshress) [inline apps/jwt_auth/tests/test_apis.py]: Tallying messages is okay and not okay. I will go with @thenav56.

Comment (thenav56) [reply apps/jwt_auth/tests/test_apis.py]: Yeah, Line 51 is sufficient.

File: apps/user/tests/test_password.py (hunk 1 above)
Comment (thenav56) [reply apps/user/tests/test_password.py]: @k9845 is referencing from this test https://github.com/django/django/blob/5fcfe5361e5b8c9738b1ee4c1e9a6f293a7dda40/tests/auth_tests/test_validators.py#L98-L117

Yeah, the message can change, so the code sounds good.

File: deep/settings.py [hunk 4]
```diff
@@ -198,6 +198,9 @@
         'NAME': 'django.contrib.auth.password_validation.'
         'NumericPasswordValidator',
     },
+    {
+        'NAME': 'user.validators.CustomMaximumLengthValidator',
```
Comment (thenav56) [inline deep/settings.py:203]: Could you add a note that we are using the Django admin panel for password reset/change here?

**Review by annshress — APPROVED** [review]

File: apps/user/tests/test_password.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
```
Comment (thenav56) [inline apps/user/tests/test_password.py]: @k9845 https://github.com/the-deep/server/pull/624/files#r548377623 This is for removing this line.
```suggestion
```

**Review by annshress — APPROVED** [review]

**Review by thenav56 — APPROVED** [review]

**Pull request merged**